### PR TITLE
Swing callback for Calendar and WeekCalendar

### PR DESCRIPTION
**Pull request by anvukelic**
Issue - https://github.com/boguszpawlowski/ComposeCalendar/issues/104

### CODE REVIEW

File: library/src/main/java/io/github/boguszpawlowski/composecalendar/Calendar.kt [hunk 1]
```diff
@@ -80,6 +81,7 @@ public fun SelectableCalendar(
   monthContainer: @Composable (content: @Composable (PaddingValues) -> Unit) -> Unit = { content ->
     Box { content(PaddingValues()) }
   },
+  swingCallback: (YearMonth) -> Unit,
```
Comment (boguszpawlowski) [inline library/src/main/java/io/github/boguszpawlowski/composecalendar/Calendar.kt:84]: In other places we name this gesture `swipe`. Also the postfix `callback` is pretty redundant - we see that it is a callback via the function signature. So I would change it to `onMonthSwipe`.

**Review by boguszpawlowski — COMMENTED** [review]
Hi @anvukelic. Thank you for your contribution. Im currently on vacation, so I wont be able to properly review this. In the meantime, please adjust the pr name / description to reference issues resolved by it.

File: library/src/main/java/io/github/boguszpawlowski/composecalendar/Calendar.kt [hunk 2]
```diff
@@ -198,19 +204,27 @@ public fun <T : SelectionState> Calendar(
...
+        MonthContent(
+          modifier = Modifier.fillMaxWidth(),
+          currentMonth = calendarState.monthState.currentMonth,
+          showAdjacentMonths = showAdjacentMonths,
+          selectionState = calendarState.selectionState,
+          today = today,
+          daysOfWeek = daysOfWeek,
+          dayContent = dayContent,
+          monthContainer = monthContainer,
+        )
```
Comment (boguszpawlowski) [inline library/src/main/java/io/github/boguszpawlowski/composecalendar/Calendar.kt:226]: I would say that this is unrelated change so probably should land in separate pr. Also, I didn't see any kind of issue / feature request for such a functionality. I think if we want to introduce something like this, and its not based on user feedback, we would have to add a boolean flag for opt-in this feature.

**Comment by anvukelic** [comment]
> Hi @anvukelic. Thank you for your contribution. Im currently on vacation, so I wont be able to properly review this. In the meantime, please adjust the pr name / description to reference issues resolved by it.

Thank you for your reply, it is not urgent, just do your vacation properly :)

In the meantime, I made the changes for the comments you've left

I raised an issue for the week names to make them not scrollable. I'll do the change in the separate PR

File: library/src/main/java/io/github/boguszpawlowski/composecalendar/week/Week.kt [hunk 3]
```diff
@@ -40,5 +40,15 @@ public data class Week(
...
     }
+
+    public fun of(
+      date: LocalDate,
+      firstDayOfWeek: DayOfWeek = WeekFields.of(Locale.getDefault()).firstDayOfWeek,
+    ): Week {
+      val offset = date.dayOfWeek.daysUntil(firstDayOfWeek)
+      val firstDay = date.minusDays(offset.toLong())
+
+      return Week(firstDay)
```
[the remark this answers is not in the document]
Comment (anvukelic) [reply library/src/main/java/io/github/boguszpawlowski/composecalendar/week/Week.kt:51]: Done

**Comment by boguszpawlowski** [comment]
@anvukelic Please check out my comment on #104
